File: challenges/2026-08-22-longest-doubling-chain/solution.go
package main

import "fmt"

func longestDoublingChain(nums []int) int {
	// TODO: implement
	return 0
}

type testCase struct {
	nums     []int
	expected int
}

func main() {
	tests := []testCase{
		{[]int{1, 2, 4, 8, 3, 10}, 4},
		{[]int{-2, -4, -8, -1}, 4},
		{[]int{0, 0, 5}, 1},
		{[]int{7}, 1},       // extra: single element
		{[]int{5, 5, 5}, 1}, // extra: all duplicates, odd, no successor
	}

	for _, tc := range tests {
		got := longestDoublingChain(tc.nums)
		status := "FAIL"
		if got == tc.expected {
			status = "PASS"
		}
		fmt.Printf("%s nums=%v expected=%d got=%d\n", status, tc.nums, tc.expected, got)
	}
}

File: challenges/2026-08-22-longest-doubling-chain/solution.py
from typing import List


def longest_doubling_chain(nums: List[int]) -> int:
    """Return the length of the longest doubling chain formable from nums."""
    # TODO: implement
    pass


if __name__ == "__main__":
    tests = [
        ([1, 2, 4, 8, 3, 10], 4),
        ([-2, -4, -8, -1], 4),
        ([0, 0, 5], 1),
        ([7], 1),          # extra: single element
        ([5, 5, 5], 1),    # extra: all duplicates, odd, no successor
    ]

    for nums, expected in tests:
        got = longest_doubling_chain(nums)
        status = "PASS" if got == expected else "FAIL"
        print(f"{status} nums={nums} expected={expected} got={got}")
